bpf/process/bpf_execve_event.c

```diff
@@ -191,11 +192,16 @@ read_exe(struct task_struct *task, struct heap_exe *exe)
 	// buffer used by d_path_local can contain up to MAX_BUF_LEN i.e. 4096 we
 	// only keep the first 255 chars for our needs (we sacrifice one char to the
 	// verifier for the > 0 check)
-	if (exe->len > 255)
-		exe->len = 255;
+	if (exe->len > BINARY_PATH_MAX_LEN - 1) {
+		offset = exe->len - (BINARY_PATH_MAX_LEN - 1);
+		exe->len = BINARY_PATH_MAX_LEN - 1;
+	}
```

Comment:
Following on with the length of postfix, I think we need to move the setting of offset into a following check (i.e. one if block for normal exe->len setting, and another new if block for offset and revlen setting):
```
__u64 revlen = exe->len;
if (exe->len > STRING_POSTFIX_MAX_LENGTH - 1) {
    offset = exe->len - (STRING_POSTFIX_MAX_LENGTH - 1);
    revlen = STRING_POSTFIX_MAX_LENGTH - 1;
}

Reply:
Could maybe move the revlen declaration (I just put it there in the example but probably needs to go up the top after offset) to avoid the static check errors.